multiview/cluster/mv_k_means.py

```diff
@@ -0,0 +1,344 @@
+        while(iter_stall < self._patience and iter_num < self._max_iter):
+            iter_num += 1
+            pre_view = (iter_num) % 2
+            view = (iter_num + 1) % 2
+
+            # Switch partitions and compute maximization
+            new_centers = list()
+            for cl in range(self._n_clusters):
+                # Isolate data points from each cluster to recompute centroids
+                mask = (partitions[pre_view] == cl)
+                if (np.sum(mask) == 0):
+                    new_centers.append(self._centroids[view][cl])
+                else:
+                    cent = np.mean(Xs[view][mask], axis=0)
+                    new_centers.append(cent)
+            self._centroids[view] = np.vstack(new_centers)
+
+            # Compute expectation
+            distances = self._compute_distance(Xs[view], self._centroids[view])
+            partitions[view] = np.argmin(distances, axis=0).flatten()
+
+            # Recompute the objective function
+            o_funct = 0
+            for cl in range(self._n_clusters):
+                # Collect data points in each cluster and compute within
+                # cluster distances
+                vecs = Xs[view][(partitions[view] == cl)]
+                dist = np.linalg.norm(vecs - self._centroids[view][cl], axis=1)
+                o_funct += np.sum(dist)
+
+            # Track of number of iterations without improvement
+            if(o_funct < objective[view]):
+                objective[view] = o_funct
+                iter_stall = 0
+            else:
+                iter_stall += 1
+
+        # Compute final cluster centroids
+        self._final_centroids(Xs)
+
+        return self
+
+    def predict(self, Xs):
+
+        '''
+        Predict the cluster labels for the data.
+
+        Parameters
+        ----------
+        Xs : list of array_likes
+            - Xs shape: (2,)
+            - Xs[0] shape: (n_samples, n_features_i)
+            This list must be of size 2, corresponding to the two
+            views of the data. The two views can each have a different
+            number of features, but they must have the same number of samples.
+
+        Returns
+        -------
+        predictions : array-like, shape (n_samples,)
+            The predicted cluster labels for each sample.
+
+        '''
+
+        Xs = check_Xs(Xs, enforce_views=2)
+
+        if self._centroids is None:
+            msg = 'This MultiviewKMeans instance is not fitted yet.'
+            raise NotFittedError(msg)
+
+        if self._centroids[0] is None:
+            msg = 'This MultiviewKMeans instance has no cluster centroids.'
+            raise AttributeError(msg)
+
+        dist1 = self._compute_distance(Xs[0], self._centroids[0])
+        dist2 = self._compute_distance(Xs[1], self._centroids[1])
+        dist_metric = dist1 + dist2
+        predictions = np.argmin(dist_metric, axis=0).flatten()
+
+        return predictions
+
+    def fit_predict(self, Xs):
```

Comment:
Should this go in base class?

Reply:
Do you mean the implemented method or the abstract method?